crates/libs/metadata/src/reader/mod.rs

```diff
@@ -103,10 +103,10 @@ pub enum InterfaceKind {
     Base,
 }
 
-#[derive(PartialEq, Eq)]
+#[derive(Copy, Clone, PartialEq, Eq)]
 pub enum SignatureKind {
-    Query,
-    QueryOptional,
+    Query((usize, usize)),
+    QueryOptional((usize, usize)),
```

Comment:
Nit: it might be more clear to give these `usize`s field names to indicate what they do.

Reply:
done